httptransport/notificationshandler.go

```diff
@@ -0,0 +1,160 @@
+}
+
+type NotifHandler struct {
+	serv service.Service
+}
+
+func NotificationHandler(serv service.Service) *NotifHandler {
+	return &NotifHandler{
+		serv: serv,
+	}
+}
+
+func (h *NotifHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+		h.Get(w, r)
+	case http.MethodDelete:
+		h.Delete(w, r)
+	default:
+		resp := &je.Response{
+			Code:    "method-not-allowed",
+			Message: "endpoint only allows POST",
+		}
+		je.Error(w, resp, http.StatusMethodNotAllowed)
+		return
+	}
+}
+
+func (h *NotifHandler) Delete(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := zerolog.Ctx(ctx)
+	path := r.URL.Path
+	id := filepath.Base(path)
+	notificationID, err := uuid.Parse(id)
+	if err != nil {
+		resp := &je.Response{
+			Code:    "bad-request",
+			Message: fmt.Sprintf("could not parse notification id: %v", err),
+		}
+		log.Warn().Err(err).Msg("could not parse notification id")
+		je.Error(w, resp, http.StatusBadRequest)
+		return
+	}
+
+	err = h.serv.DeleteNotifications(ctx, notificationID)
+	if err != nil {
+		resp := &je.Response{
+			Code:    "internal-server-error",
+			Message: fmt.Sprintf("could not delete notification: %v", err),
+		}
+		log.Warn().Err(err).Msg("could not delete notification")
+		je.Error(w, resp, http.StatusInternalServerError)
+		return
+	}
+	return
+}
+
+// NotificaitonsHandler will return paginated notifications to the caller.
+func (h *NotifHandler) Get(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := zerolog.Ctx(ctx)
+	path := r.URL.Path
+	id := filepath.Base(path)
+	notificationID, err := uuid.Parse(id)
+	if err != nil {
+		resp := &je.Response{
+			Code:    "bad-request",
+			Message: fmt.Sprintf("could not parse notification id: %v", err),
+		}
+		log.Warn().Err(err).Msg("could not parse notification id")
+		je.Error(w, resp, http.StatusBadRequest)
+		return
+	}
+
+	// optional page_size parameter
+	var pageSize uint64
+	if param := r.URL.Query().Get("page_size"); param != "" {
+		pageSize, err = strconv.ParseUint(param, 10, 64)
+		if err != nil {
+			resp := &je.Response{
+				Code:    "bad-request",
+				Message: "could not parse \"page_size\" query param into integer",
+			}
+			je.Error(w, resp, http.StatusBadRequest)
+			return
+		}
+	}
+	if pageSize == 0 {
+		pageSize = DefaultPageSize
+	}
+
+	// optional page parameter
+	var page int64
+	if param := r.URL.Query().Get("page"); param != "" {
+		page, err = strconv.ParseInt(param, 10, 64)
+		if err != nil {
+			resp := &je.Response{
+				Code:    "bad-request",
+				Message: "could not parse \"page\" query param into uuid",
+			}
+			je.Error(w, resp, http.StatusBadRequest)
+			return
+		}
+	}
+	// downstream code protects against this, but just to be defensive
+	// pages are not 0 indexed.
+	if page == 0 {
+		page = 1
+	}
+
+	inP := &pager.Page{
+		Size: pageSize,
+		Next: page,
+	}
+	notifications, outP, err := h.serv.Notifications(ctx, notificationID, inP)
```

Comment:
Does this mean it's impossible to call without paging?

Reply:
Yes, the API assumes paging.